libr/bin/p/bin_mdt.c

```diff
@@ -0,0 +1,532 @@
+					eo->ehdr.e_ident[0], eo->ehdr.e_ident[1], eo->ehdr.e_ident[2], eo->ehdr.e_ident[3],
+					eo->ehdr.e_ident[4], eo->ehdr.e_ident[5], eo->ehdr.e_ident[6], eo->ehdr.e_ident[7],
+					eo->ehdr.e_ident[8], eo->ehdr.e_ident[9], eo->ehdr.e_ident[10], eo->ehdr.e_ident[11],
+					eo->ehdr.e_ident[12], eo->ehdr.e_ident[13], eo->ehdr.e_ident[14], eo->ehdr.e_ident[15]);
+				bf->rbin->cb_printf ("0x00000010  Type        0x%04x\n", eo->ehdr.e_type);
+				bf->rbin->cb_printf ("0x00000012  Machine     0x%04x\n", eo->ehdr.e_machine);
+				bf->rbin->cb_printf ("0x00000014  Version     0x%08x\n", eo->ehdr.e_version);
+				bf->rbin->cb_printf ("0x00000018  Entrypoint  0x%08"PFMT64x"\n", (ut64)eo->ehdr.e_entry);
+				bf->rbin->cb_printf ("0x0000001c  PhOff       0x%08"PFMT64x"\n", (ut64)eo->ehdr.e_phoff);
+				bf->rbin->cb_printf ("0x00000020  ShOff       0x%08"PFMT64x"\n", (ut64)eo->ehdr.e_shoff);
+				bf->rbin->cb_printf ("0x00000024  Flags       0x%04x\n", eo->ehdr.e_flags);
+				bf->rbin->cb_printf ("0x00000028  EhSize      %d\n", eo->ehdr.e_ehsize);
+				bf->rbin->cb_printf ("0x0000002a  PhentSize   %d\n", eo->ehdr.e_phentsize);
+				bf->rbin->cb_printf ("0x0000002c  PhNum       %d\n", eo->ehdr.e_phnum);
+				bf->rbin->cb_printf ("0x0000002e  ShentSize   %d\n", eo->ehdr.e_shentsize);
+				bf->rbin->cb_printf ("0x00000030  ShNum       %d\n", eo->ehdr.e_shnum);
+				bf->rbin->cb_printf ("0x00000032  ShStrndx    %d\n", eo->ehdr.e_shstrndx);
+				bf->rbin->cb_printf (" --- ELF HEADER END --- \n\n");
+			} else {
+				bf->rbin->cb_printf (" ------- FAILED ------- \n");
+			}
+			break;
+		case R_BIN_MDT_PART_MBN:
+			bf->rbin->cb_printf (" | MBN signature segment\n");
+			if (part->obj.mbn) {
+				SblHeader *mbn = (SblHeader *)part->obj.mbn;
+				bf->rbin->cb_printf (" -- MBN AUTH HEADER BEGIN -- \n");
+				bf->rbin->cb_printf ("0x00 image_id:   kMbnImageNone (0x%x)\n", 0);
+				bf->rbin->cb_printf ("0x04 version:    0x%x\n", mbn->version);
+				bf->rbin->cb_printf ("0x08 paddr:      0x%x\n", mbn->paddr);
+				bf->rbin->cb_printf ("0x0c vaddr:      0x%x\n", mbn->vaddr);
+				bf->rbin->cb_printf ("0x10 psize:      0x%x\n", mbn->psize);
+				bf->rbin->cb_printf ("0x14 code_pa:    0x%x\n", mbn->code_pa);
+				bf->rbin->cb_printf ("0x18 sign_va:    0x%x\n", mbn->sign_va);
+				bf->rbin->cb_printf ("0x1c sign_sz:    0x%x\n", mbn->sign_sz);
+				bf->rbin->cb_printf ("0x20 cert_va:    0x%x\n", mbn->cert_va);
+				bf->rbin->cb_printf ("0x24 cert_sz:    0x%x\n", mbn->cert_sz);
+				bf->rbin->cb_printf (" --- MBN AUTH HEADER END --- \n\n");
+			} else {
+				bf->rbin->cb_printf (" ------- FAILED ------- \n");
+			}
+			break;
+		}
+		i++;
+	}
+}
+
+static void mdt_map_free(void *ptr) {
+	RBinMap *map = (RBinMap *)ptr;
+	if (map) {
+		free (map->file);
+		free (map);
+	}
+}
+
+static RBinSection *segment_to_section(ut64 paddr, ut64 vaddr, ut64 psize, ut64 vsize, ut32 flags, const char *name) {
+	RBinSection *section = R_NEW0 (RBinSection);
```

Comment:
add an R_RETURN_VAL_IF_FAIL check for name and tag the argument as R_NONNULL so the compiler can chk at compile time